lib/Analysis/DependenceAnalysis.cpp

```diff
@@ -131,17 +131,35 @@ circt::analysis::MemoryDependenceAnalysis::MemoryDependenceAnalysis(
   // Collect affine loops grouped by nesting depth.
   std::vector<SmallVector<AffineForOp, 2>> depthToLoops;
   mlir::affine::gatherLoops(funcOp, depthToLoops);
+  depthToLoops.erase(depthToLoops.begin());
```

Comment:
I'm curious why this is needed, since it needs special handling below.

Reply:
It should make the `isAncestor()` checks faster, since we know none of the root for loops will be ancestors of one another.

Reply:
Personally, I don't think removing the first element of this vector is worth it, since it's mildly confusing why we want to discard one element and have to handle that later. I don't expect there to be a major performance difference either. I would say its fine to remove this line and stick to the normal `depthToLoops` structure, but if you do want to keep this, please add a comment explaining why its there and how it changes the assumptions of what `depthToLoops` contains.